pkg/clientwrappers/parametercontext/parametercontext.go

```diff
@@ -365,3 +375,16 @@ func updateRequest2Status(updateRequest *nigoapi.ParameterContextUpdateRequestEn
 		NotFoundRetryCount: 0,
 	}
 }
+
+func extractMaximumSecretTime(parameterSecrets []*corev1.Secret) *metav1.Time {
```

Comment:
```suggestion
func extractMostRecentSecretUpdateTime(parameterSecrets []*corev1.Secret) *metav1.Time {
```

Reply:
can you add a unit test for this method just to be certain it does what we want it to?

Reply:
Should be ok for the unit test. Let me know if you want more cases.